src/api/entities/Identity/ChildIdentity.ts

```diff
@@ -0,0 +1,80 @@
+import { UniqueIdentifiers } from '~/api/entities/Identity';
+import { unlinkChildIdentity } from '~/api/procedures/unlinkChildIdentity';
+import { Context, Identity } from '~/internal';
+import { NoArgsProcedureMethod } from '~/types';
+import { identityIdToString, stringToIdentityId } from '~/utils/conversion';
+import { createProcedureMethod } from '~/utils/internal';
+
+/**
+ * Represents a child identity
+ */
+export class ChildIdentity extends Identity {
+  /**
+   * @hidden
+   */
+  public constructor(identifiers: UniqueIdentifiers, context: Context) {
+    super(identifiers, context);
+
+    this.unlinkFromParent = createProcedureMethod(
+      {
+        getProcedureAndArgs: () => [unlinkChildIdentity, { child: this }],
+        voidArgs: true,
+      },
+      context
+    );
+  }
+
+  /**
+   * Returns the parent of this Identity (if any)
+   */
+  public async getParentDid(): Promise<Identity | null> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+      context,
+      did,
+    } = this;
+
+    const rawIdentityId = stringToIdentityId(did, context);
+    const rawParentDid = await identity.parentDid(rawIdentityId);
+
+    if (rawParentDid.isEmpty) {
+      return null;
+    }
+
+    const parentDid = identityIdToString(rawParentDid.unwrap());
+
+    return new Identity({ did: parentDid }, context);
+  }
+
+  /**
+   * @hidden
+   * since a child Identity doesn't has any other children, this method overrides the base implementation to return empty array
```

Comment:
Should it return an empty array or should it throw? Not sure about it.

Reply:
I'm OK with both, but I preferred an empty array since throw requires the end user to handle this separately (which personally I think is an overkill)